### DISCUSSION TITLE
.config/fish/fish_variables has changed since chezmoi last wrote it?

### DISCUSSION THREAD

**Discussion by wchrisjohnson** [twpayne/chezmoi#3579] [Q&A] [answered]
I have recently started trying out Chezmoi, and like it a lot. However, yesterday, after adding my `fish` shell config to Chezmoi I started seeing this:
```
$ chezmoi apply -v
.config/fish/fish_variables has changed since chezmoi last wrote it? diff/overwrite/all-overwrite/skip/quit
```

This looks similar to:
https://github.com/twpayne/chezmoi/discussions/1935

I'm not using `.chezmoiexternal.toml`. I've searched for caches in fish and the `fisher` based prompt `tide` that I'm using, to no avail. Somehow after nearly a dozen attempts to fix `fish_variables` by `chezmoi cd + git add . + git push ...` I'm at a loss.

Any ideas of what is going on?

**Comment by halostatue** [comment, marked as answer]
`fish_variables` *is* a cache file. It's where universal variables, that may be updated independently of any action you take, get stored.

You mentioned that you're using IlanCosman/tide, which means that `_tide_left_items` and `_tide_right_items` may be updated based on availability. You would need to check a chezmoi diff between the two versions to see what's changing.

I have [previously recommended](https://github.com/twpayne/chezmoi/issues/2869#issuecomment-1475569373) not syncing `fish_variables`, because there are values which will differ by machine/os. By and large, I have also started *not* using universal variables in the fish plug-ins that I write, but that's a different issue.

**Reply by wchrisjohnson** [reply]
Thanks @halostatue - I _did_ search the discussions for `fish_variables` related questions but missed the one you liked somehow; sorry to have duplicated the questions! All good now that I've ignored the file.

**Reply by halostatue** [reply]
Even if you had seen that discussion, it may not have been clear that `fish_variables` is a dynamic file, especially when you are using an advanced prompt like `tide`.
